client/extensions/woocommerce/index.js

```diff
@@ -151,15 +155,23 @@ function addStorePage( storePage, storeNavigation ) {
 	} );
 }
 
-function createStoreNavigation( context, next ) {
-	const state = context.store.getState();
-	const selectedSite = getSelectedSite( state );
+function mapStateToProps( state ) {
+	let siteId = getSelectedSiteId( state );
+	if ( ! siteId && ( 1 === getCurrentUserSiteCount( state ) ) ) {
+		siteId = getPrimarySiteId( state );
+	}
+	const site = getSite( state, siteId );
```

Comment:
Agreed - I've added the selector in https://github.com/Automattic/wp-calypso/pull/14022/commits/3c0ec04fc96a5d46464a6d123899c97b94c4878a